src/transformers/models/jetmoe/modeling_jetmoe.py

```diff
@@ -0,0 +1,1731 @@
+        )
+        gates = zeros.scatter(1, top_k_indices, 1)
+        expert_size = gates.long().sum(0)
+        top_k_gates = top_k_gates.flatten()
+        top_k_experts = top_k_indices.flatten()
+        _, index_sorted_experts = top_k_experts.sort(0)
+        batch_index = index_sorted_experts.div(self.top_k, rounding_mode="trunc")
+        batch_gates = top_k_gates[index_sorted_experts]
+        expert_size = expert_size.tolist()
+
+        return index_sorted_experts, batch_index, batch_gates, expert_size
+
+    def forward(self, x, return_topo=False):
+        """
+        Compute the top-k gating for the input.
+
+        See paper: https://arxiv.org/abs/1701.06538.
+
+        Args:
+            x (torch.Tensor):
+                Input tensor with shape [batch_size, input_size].
+
+        Returns:
+            torch.Tensor:
+                Top-k indices.
+            torch.Tensor:
+                Top-k gating values.
+            torch.Tensor:
+                router layer logits.
+        """
+
+        logits = self.layer(x).float()
+        top_k_logits, top_k_indices = logits.topk(self.top_k, dim=1)
+        top_k_gates = torch.softmax(top_k_logits, dim=1).type_as(x)
+
+        if return_topo:
+            return self.compute_topo(top_k_indices, top_k_gates) + (logits,)
+        else:
+            return top_k_indices, top_k_gates, logits
+
+
+class JetMoeMoE(nn.Module):
+    """
+    A Sparsely gated mixture of experts layer with 1-layer Feed-Forward networks as experts.
+
+    Args:
+        config:
+            Configuration object with model hyperparameters.
+    """
+
+    def __init__(self, config: JetMoeConfig):
+        super(JetMoeMoE, self).__init__()
+
+        self.num_experts = config.num_local_experts
+        self.input_size = config.hidden_size
+        self.hidden_size = config.intermediate_size
+        self.top_k = config.num_experts_per_tok
+        self.activation = ACT2FN[config.activation_function]
+        self.bias = torch.nn.Parameter(torch.empty(self.input_size))
+        self.input_linear = JetMoeParallelExperts(self.num_experts, self.input_size, self.hidden_size * 2)
+        self.output_linear = JetMoeParallelExperts(self.num_experts, self.hidden_size, self.input_size)
+
+        self.router = JetMoeTopKGating(
+            input_size=self.input_size,
+            num_experts=self.num_experts,
+            top_k=self.top_k,
+        )
+
+    def forward(self, layer_input):
+        """
+        Forward pass of the mixture of experts layer.
+
+        Args:
+            layer_input (Tensor):
+                Input tensor.
+
+        Returns:
+            Tensor:
+                Output tensor.
+            Tensor:
+                Router logits.
+        """
+        bsz, length, emb_size = layer_input.size()
+        layer_input = layer_input.reshape(-1, emb_size)
+        _, batch_index, batch_gates, expert_size, router_logits = self.router(layer_input, return_topo=True)
```

Comment:
```suggestion
        _, batch_index, batch_gates, expert_size, router_logits = self.router(layer_input)
```
See my comment above about `return_topo`